core/src/main/java/apoc/export/json/JsonImporter.java

```diff
@@ -130,11 +141,28 @@ private void manageNode(Map<String, Object> param) {
             lastLabels = labels;
         }
         if (!labels.equals(lastLabels)) {
+            checkConstraints(labels);
             flush();
             lastLabels = labels;
         }
     }
 
+    private void checkConstraints(List<String> labels) {
+        if (labels.isEmpty()) {
+            return;
+        }
+        try (final Transaction tx = db.beginTx()) {
+            final Schema schema = tx.schema();
+            final String importIdName = importJsonConfig.getImportIdName();
+            final String label = labels.get(0);
```

Comment:
Why are we only looking at the first label?

Reply:
Sorry my bad. Effectively it makes much more sense to check all the labels. Changed.